sourced/ml/transformers/repo2df.py

```diff
@@ -1,33 +1,30 @@
 from sourced.ml.transformers import Transformer
 
 
-class Repo2DocFreq(Transformer):
-    NDOCS_KEY = -1, 0
+class Documents2DocFreq(Transformer):
 
-    def __init__(self, extractors, **kwargs):
+    def __init__(self, extractors, document_column='blob_id', **kwargs):
         super().__init__(**kwargs)
         self.extractors = extractors
+        self.document_column = document_column
 
     def __call__(self, rows):
-        processed = rows.flatMap(self.process_row)
+        processed = rows.flatMap(self._process_row)
         if self.explained:
             self._log.info("toDebugString():\n%s", processed.toDebugString().decode())
-        reduced = processed.countByKey()
-        ndocs = None
+        ndocs = rows.map(lambda x: x.asDict(self.document_column)).count()
+        reduced = processed.distinct().map(lambda x: ((x[0], x[1]), 1)).countByKey()
+
         for (i, key), value in reduced.items():
-            if (i, key) == self.NDOCS_KEY:
-                ndocs = value
-                continue
             self.extractors[i].apply_docfreq(key, value)
 
         for extractor in self.extractors:
             extractor.ndocs = ndocs
 
-    def process_row(self, row):
-        yield self.NDOCS_KEY, 1
+    def _process_row(self, row):
         for i, extractor in enumerate(self.extractors):
             for k in extractor.inspect(row.uast):
-                yield (i, k), 1
```

Comment:
This breaks the upstream logic which relies on the composite key

Reply:
Not really, also that the reason for renaming. This method is used only in `__call__` method of this class.